python/lsst/pipe/base/script/transfer_from_graph.py

```diff
@@ -124,45 +123,53 @@ def transfer_from_graph(
         output_run = metadata.get("output_run")
         output = metadata.get("output")
         input = metadata.get("input")
         if output_run is not None and output is not None:
-            _update_chain(dest_butler.registry, output, output_run, input)
+            _update_chain(dest_butler, output, output_run, input)
 
     return count
 
 
-def _update_chain(registry: Registry, output_chain: str, output_run: str, inputs: list[str] | None) -> None:
+def _update_chain(butler: Butler, output_chain: str, output_run: str, inputs: list[str] | None) -> None:
     """Update chain definition if it exists to include run as the first item
     in a chain. If it does not exist then create it to include all inputs and
     output.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        Butler where to update the collection chain.
+    output_chain : `str`
+        Name of the output CHAINED collection.
+    output_run : `str`
+        Name of the output RUN collection.
+    inputs : `list` [`str`] | None
+
+
     """
+    # Do not need to update chain if output_run does not already exists.
     try:
-        # If output_chain is not a chain the exception will be raised.
-        chain_definition = list(registry.getCollectionChain(output_chain))
+        _ = butler.collections.get_info(output_run)
     except MissingCollectionError:
-        # We have to create chained collection to include inputs and output run
-        # (this reproduces logic in CmdLineFwk).
-        registry.registerCollection(output_chain, type=CollectionType.CHAINED)
-        chain_definition = list(registry.queryCollections(inputs, flattenChains=True)) if inputs else []
-        chain_definition = [output_run] + [run for run in chain_definition if run != output_run]
-        registry.setCollectionChain(output_chain, chain_definition)
-    else:
-        # If run is in the chain but not the first item then remove it, will
-        # re-insert at front below.
-        try:
-            index = chain_definition.index(output_run)
-            if index == 0:
-                # It is already at the top.
-                return
-            else:
-                del chain_definition[index]
-        except ValueError:
-            pass
-
-        chain_definition.insert(0, output_run)
-        registry.setCollectionChain(output_chain, chain_definition)
+        _LOG.verbose(
+            "output RUN collection (%s) does not exist.  Skipping updating the output chain.",
+            output_run,
+        )
+        return
+
+    # Make chain collection if doesn't exist before calling prepend_chain.
+    created_now = butler.collections.register(output_chain, CollectionType.CHAINED)
+    if created_now:
+        _LOG.verbose("Registered chain collection: %s", output_chain)
+        if inputs:
+            # Add input collections to chain collection just made.
+            butler.collections.prepend_chain(output_chain, inputs)
```

Comment:
I think if we are racing with another caller, we probably want to use extend_chain here so that the inputs end up on the end rather than risk them jumping in front of a RUN output collection.